components/CompanyRegister.tsx

```diff
@@ -1,128 +1,145 @@
 import { CreateCompanyItem } from '@/db/entities/company';
 import useCompany from '@/hooks/useCompanyRegistration';
 import { isSuccessResponse } from '@/pages/api/companies';
+import { zodResolver } from '@hookform/resolvers/zod';
 import { useRouter } from 'next/router';
 import { useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
 import { toast } from 'react-toastify';
 import { useAccount } from 'wagmi';
+import { z } from 'zod';
 import { useRecruitmentRegisterCompany } from '../src/generated';
 import { Badge } from './ui/badge';
 
+const schema = z.object({
+  companyName: z
+    .string()
+    .min(1, { message: 'Company Name is required' })
+    .max(255),
+  companyEmail: z.string().email({ message: 'Invalid email format' }),
+});
+
+interface FormData {
+  companyName: string;
+  companyEmail: string;
+  companySite: string;
+}
+
 const CompanyRegister = () => {
   const router = useRouter();
   const { address } = useAccount();
-  const { isValidating, isRegistered } = useCompany(address);
+  const { isRegistered } = useCompany(address);
+  const [isValidating] = useState(false);
   const { isLoading, isSuccess, writeAsync } = useRecruitmentRegisterCompany();
 
-  const [companyName, setCompanyName] = useState('');
-  const [companyEmail, setCompanyEmail] = useState('');
-  const [companySite, setCompanySite] = useState('');
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isValid },
+    getValues,
+  } = useForm<FormData>({ resolver: zodResolver(schema) });
 
-  const registerCompanySc = async () => {
-    if (isRegistered) {
+  const onSubmit = async (data: FormData) => {
+    if (!isRegistered) {
+      try {
+        await writeAsync();
+        registerCompanyDb(data);
+      } catch (error) {
+        console.error('Error:', error);
+        toast.error('Registration Unsuccessful. Please contact support.');
+      }
+    } else {
       toast.error('Wallet address is already registered');
-      return;
-    }
-    try {
-      await writeAsync();
-    } catch (e) {
-      toast.error('Company registration failed. Please contact support');
     }
   };
 
-  const registerCompanyDb = async () => {
-    const companyData: CreateCompanyItem = {
-      companyId: address as string,
-      name: companyName,
-      email: companyEmail,
-      site: companySite,
-    };
-    const options = {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify(companyData),
-    };
+  const registerCompanyDb = async (data: FormData) => {
+    const validatedData = schema.safeParse(data);
```

Comment:
the form can not be submitted unless valid so is this an unneeded check?

Reply:
yeah truly adjusted.